### PR TITLE
 curriculum: Change pre and code to blockquote in Advanced Node and Express challenges

### PR DESCRIPTION
**Pull request by thecodingaviator**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Refer #35697 #35709

### CODE REVIEW

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md [hunk 1]
```diff
@@ -9,17 +9,21 @@ challengeType: 2
...
-      return next();
-  }
-  res.redirect('/');
-};</pre>
+<blockquote>
+function ensureAuthenticated(req, res, next) {<br>
+&nbsp;&nbsp;if (req.isAuthenticated()) {<br>
+&nbsp;&nbsp;&nbsp;&nbsp;return next();<br>
+&nbsp;&nbsp;}<br>
+&nbsp;&nbsp; res.redirect('/');<br>
```
Comment (RandellDawson) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]:
```suggestion
&nbsp;&nbsp;res.redirect('/');<br>
```

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md [hunk 2]
```diff
@@ -9,17 +9,21 @@ challengeType: 2
...
+};
+</blockquote>
 Now add <em>ensureAuthenticated</em> as a middleware to the request for the profile page before the argument to the get request containing the function that renders the page.
-<pre>app.route('/profile')
-  .get(ensureAuthenticated, (req,res) => {
-       res.render(process.cwd() + '/views/pug/profile');
-  });</pre>
+<blockquote>
+app.route('/profile')<br>
```
Comment (RandellDawson) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]: I also debate whether or not to indent the method (chained or not).  Would either of the following be more readable?
```
app.route('/profile').get(
  ensureAuthenticated,
  (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);

```
OR not indenting `.get`
```
app.route('/profile')
.get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my suggestions and also my comments below.

For Registration of New Users, should this be two-space indentation (more like below)?
```
app.route('/register')
  .post((req, res, next) => {
    db.collection('users').findOne({ username: req.body.username }, function(err, user) {
      if (err) {
        next(err);
      } else if (user) {
        res.redirect('/');
      } else {
        db.collection('users').insertOne({
          username: req.body.username,
          password: req.body.password
        },
          (err, doc) => {
            if (err) {
              res.redirect('/');
            } else {
              next(null, user);
            }
          }
        )
      }
    })
  },
    passport.authenticate('local', { failureRedirect: '/' }),
    (req, res, next) => {
      res.redirect('/profile');
    }
  );
```

The same question for Implementation of Social Authentication II.
```
passport.use(new GitHubStrategy({
  clientID: process.env.GITHUB_CLIENT_ID,
  clientSecret: process.env.GITHUB_CLIENT_SECRET,
  callbackURL: /*INSERT CALLBACK URL ENTERED INTO GITHUB HERE*/
},
  function(accessToken, refreshToken, profile, cb) {
    console.log(profile);
    //Database logic here with callback containing our user object
  }
));
```

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md (hunk 2 above)
Comment (thecodingaviator) [reply curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]: I guess not indenting looks better

Comment (thecodingaviator) [reply curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]: Rather, how doeds this look:
```
app
  .route('/profile')
  .get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);

**Comment by thecodingaviator** [comment]
About the Registration of new users, I'm fine either way I guess, don't really see how it'll affect anything. What do you think?

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md (hunk 2 above)
Comment (RandellDawson) [reply curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]: @thecodingaviator Go with the last one above.

**Comment by RandellDawson** [comment]
> About the Registration of new users, I'm fine either way I guess, don't really see how it'll affect anything. What do you think?

I like the two space indention that I recommended above for the `Registration of New Users` challenge.

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md (hunk 2 above)
Comment (thecodingaviator) [reply curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]: The one I posted or the latter one you said, I’m sorry a bit confused

Comment (RandellDawson) [reply curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]:
```
app
  .route('/profile')
  .get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);
```
Your version.

**Comment by ojeytonwilliams** [comment]
Sorry to ask this so late in the process, but should we be making them into `<blockquote>`s?  These are code blocks, so the semantic markup is to wrap `<code>` with `<pre>`.

**Comment by thecodingaviator** [comment]
@RandellDawson How does it look now?

**Comment by thecodingaviator** [comment]
@RandellDawson My bad, how does it look now?

**Comment by ojeytonwilliams** [comment]
@RandellDawson Yeah, you're quite right - this wasn't the place to talk about it.  I've created issue #35913 to address it.

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md [hunk 3]
```diff
@@ -10,32 +10,37 @@ As a reminder, this project is being built upon the following starter project on
...
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}<br>
+&nbsp;&nbsp;&nbsp;&nbsp;})<br>
+&nbsp;&nbsp;},<br>
+&nbsp;&nbsp;&nbsp;&nbsp;passport.authenticate('local', { failureRedirect: '/' }),<br>
+&nbsp;&nbsp;&nbsp;&nbsp;(req, res, next) => {<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;res.redirect('/profile');<br>
+&nbsp;&nbsp;&nbsp;&nbsp;}<br>
+&nbsp;&nbsp;);<br>
```
Comment (RandellDawson) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md]:
```suggestion
&nbsp;&nbsp;);
```

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md [hunk 4]
```diff
@@ -10,32 +10,37 @@ As a reminder, this project is being built upon the following starter project on
...
+);
```
Comment (RandellDawson) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md]:
```suggestion
```

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md [hunk 5]
```diff
@@ -9,17 +9,23 @@ challengeType: 2
...
-<pre>app.route('/profile')
-  .get(ensureAuthenticated, (req,res) => {
-       res.render(process.cwd() + '/views/pug/profile');
-  });</pre>
+<blockquote>
+app<br>
+&nbsp;.route('/profile')<br>
+&nbsp;.get(ensureAuthenticated, (req,res) => {<br>
+&nbsp;&nbsp;res.render(process.cwd() + '/views/pug/profile');<br>
+&nbsp;}<br>
```
Comment (RandellDawson) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]:
```suggestion
&nbsp;});
```

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md [hunk 6]
```diff
@@ -9,17 +9,23 @@ challengeType: 2
...
+);
```
Comment (RandellDawson) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md]:
```suggestion
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@thecodingaviator I have made two suggestions on two challenges. 

1. Create New Middle - Needed to have the `);` moved up to match the indentation of the `.route` where the `(` originated.

2. Registration of New Users - The suggestion corrects a syntax error that was present in the code.

We these two changes made, we can get this merged.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@thecodingaviator Thanks for making those change.  LGTM

**Comment by ojeytonwilliams** [comment]
@thecodingaviator Thank you for your contribution to the page! :+1:
We're happy to accept these changes, and look forward to future contributions. :memo:

**Pull request merged**